src/createWebStoragePersister/index.ts

```diff
@@ -31,7 +33,7 @@ export function createWebStoragePersister({
   //try to save data to storage
   function trySave(persistedClient: PersistedClient) {
     try {
-      storage.setItem(key, serialize(persistedClient))
+      storage?.setItem(key, serialize(persistedClient))
```

Comment:
can we move the whole function into the `if (typeof storage !== 'undefined') {` check and make it an arrow function? That way, we can closure over `storage` without having to check for `undefined` again with optional chaining.

Reply:
i totally forgot that I also tackled some of these changes in a different PR for v4 🙈 
- https://github.com/tannerlinsley/react-query/pull/3556

anyhow, conflicts shouldn't be severe.